taskvine/src/worker/vine_cache.c

```diff
@@ -376,6 +380,46 @@ static int do_transfer( struct vine_cache *c, const char *source_url, const char
 	return result;
 }
 
+/* 
+scan through transfer processes and mark them as completed as they finish.
+*/
+
+int vine_cache_wait(struct vine_cache *c)
+{
+	int status;
+	int exit_code;
+	struct cache_file *f;
+    	char * cachename;
+	HASH_TABLE_ITERATE(c->table, cachename, f){
+		if(f->pid && !f->complete){
+			int result = waitpid(f->pid, &status, WNOHANG);
+			if(result==0){
+				// pid is still going
+			}
+			else if(result<0){
+				debug(D_VINE, "wait4 on pid %d returned an error: %s",(int)f->pid,strerror(errno));	
+			}
+			else if(result>0){
+				if(!WIFEXITED(status)){
+					exit_code = WTERMSIG(status);
+					debug(D_VINE, "transfer process (pid %d) exited abnormally with signal %d",f->pid, exit_code);
+					f->complete = -1;
```

Comment:
Another case where a variable looks like a boolean, but isn't.
Make this an enumeration that explains what all the possible values mean.